netket/utils/group/_permutation_group.py

```diff
@@ -23,35 +23,79 @@
 from netket.utils.types import Array, DType, Shape
 from netket.utils.dispatch import dispatch
 
+from netket.utils import warn_deprecation, deprecated
+
 from ._group import FiniteGroup
 from ._semigroup import Element
 
 
 class Permutation(Element):
-    def __init__(self, permutation: Array, name: str | None = None):
+    def __init__(
+        self,
+        permutation: Array | None = None,
+        *,  # change one line somewhere
+        name: str | None = None,
+        permutation_array: Array | None = None,
+        inverse_permutation_array: Array | None = None,
+    ):
         r"""
-        Creates a `Permutation` from an array of preimages of :code:`range(N)`.
+        Creates a `Permutation` from either the array of images `permutation_array` or preimages `inverse_permutation_array`. Note that the left action of a permutation on an array `a` is `a[inverse_permutation_array]`.
```

Comment:
Break up to lines of up to 70-80 characters as per PEP 8. Also, the first tagline should be no more than 1-2 of these lines, because it makes the autogenerated documentation weird. The "Note that" bit can come after an empty line - and probably expanded.